src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -415,6 +453,9 @@ def _generate_components_list(self):
         self._components_efficiency = components_efficiency
         self._components_control_parameters = components_control_parameter
 
+        # Populate cache
+        self._set_cache_instance(COMPONENT_VARIABLE)
+
     def _get_connections(self):
```

Comment:
While we are at it: shouldn't the content of the _get_connections() method be put into a different function like _generate_connections_list() so that we can do as for the the get_component() method ?